### PR TITLE
feat(forms): onTrigger !== onChange

### PR DESCRIPTION
**Pull request by acateland**
**Please check if the PR fulfills these requirements**

- [x] The commit(s) message(s) follows our
  [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
onChange callback was really mapped to a onTrigger behavior.

**What is the new behavior?**
onChange now map to the underlying component onChange behavior
onTrigger is created to handle solely onTrigger behavior

**Does this PR introduce a breaking change?**

- [X] Yes
- [ ] No

onChange actual behavior was linked to onTrigger property since it handled only trigger execution.
onChange is mapped back to the underlying https://github.com/mozilla-services/react-jsonschema-form lib onChange behavior.

Better change all your Form.onChange to onTrigger

**Other information**:

### CODE REVIEW

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://364.talend.surge.sh)

File: packages/forms/src/Form.js [hunk 1]
```diff
@@ -76,6 +77,7 @@ class Form extends React.Component {
 
 	/**
 	 * Handle changes only if modified field has "ui:trigger" option
+	 * use onTrigger property to take asvantage of this feature
```
Comment (frassinier) [inline packages/forms/src/Form.js]: advantage*

File: packages/forms/BREAKING_CHANGES_LOG.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
+# Breaking changes log
+
+Before 1.0, `react-talend-forms` do NOT follow semver version in releases.
+This document aims to ease the WIP migration from a version to another by providing intels about what to do to migrate.
+
+## v0.71.0
+
+* PR #364 [chore: remove Icon component]
```
Comment (jmfrancois) [inline packages/forms/BREAKING_CHANGES_LOG.md]: wrong title

File: packages/forms/BREAKING_CHANGES_LOG.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+	onSubmit={this.onSubmit}
+	actions={this.formActions(this.props.definition, this.props.formData.label, onCancelAction)}
+	buttonBlockClass={buttonBlockClass}
+/>
+```
+Can even use `real` onChange
+```javascript
+<Form
+	data={schema}
+    onChange={this.onChange}
```
Comment (jmfrancois) [inline packages/forms/BREAKING_CHANGES_LOG.md]: tabs , no space

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://364.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://364.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://364.talend.surge.sh)

**Comment by jmfrancois** [comment]
:octocat: [Demo is available here](http://364.talend.surge.sh)

**Pull request merged**